### ISSUE TITLE
junk line in PDF/DOCX output: --- title: datatables header-include: |- head: '' background-color: white ---

### ISSUE THREAD

**Issue by bbest** [rstudio/DT#471]
Hi @yihui,

## Problem

Thank you for this wonderful package! I am applying some tricky formatting using SVG colored bars in the table cells that worked well with printing DT::datatable() to PDF (and excludes more typical PDF table functions), but recently I'm seeing this extra text showing up at the top of the created intermediary PDF figure that gets embedded in the main tex document:

```
--- title: datatables header-include: |- head: '' background-color: white ---
```

BTW, this also shows up in the embedded PNG when knitting to `word_document`.

## Reproducible Example

Here's a super simple reproducible example for an `test-dt.Rmd`:

```yaml
---
output: pdf_document
---
```

```{r}
library(DT)
datatable(mtcars)
```

```{r}
devtools::session_info()
```

When this gets knitted I see in the PDF:

![image](https://user-images.githubusercontent.com/2837257/33901900-1c35e2a8-df28-11e7-81f0-a0b077e1933c.png)

## Debugging

I can't tell if this extra text is coming from the `DT`, `rkmarkdown`, or `knitr` packages. Simply using the `keep_tex: true` is not helpful since the output PDF of the table is seperately generated as PDF and included in the main tex document.

### Packages with Error

Here's my `session_info()` on a Mac using latest CRAN package updates that produces the erroneous text:

```
## Session info -------------------------------------------------------------
##  setting  value                       
##  version  R version 3.4.0 (2017-04-21)
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       Etc/UTC                     
##  date     2017-12-13
## Packages -----------------------------------------------------------------
##  package     * version date       source        
##  backports     1.1.0   2017-05-22 CRAN (R 3.4.0)
##  base        * 3.4.0   2017-06-05 local         
##  compiler      3.4.0   2017-06-05 local         
##  datasets    * 3.4.0   2017-06-05 local         
##  devtools      1.13.2  2017-06-02 CRAN (R 3.4.0)
##  digest        0.6.12  2017-01-27 CRAN (R 3.4.0)
##  DT          * 0.2     2016-08-09 CRAN (R 3.4.0)
##  evaluate      0.10    2016-10-11 CRAN (R 3.4.0)
##  graphics    * 3.4.0   2017-06-05 local         
##  grDevices   * 3.4.0   2017-06-05 local         
##  htmltools     0.3.6   2017-04-28 CRAN (R 3.4.0)
##  htmlwidgets   0.8     2016-11-09 CRAN (R 3.4.0)
##  jsonlite      1.5     2017-06-01 CRAN (R 3.4.0)
##  knitr         1.16    2017-05-18 CRAN (R 3.4.0)
##  magrittr      1.5     2014-11-22 CRAN (R 3.4.0)
##  memoise       1.1.0   2017-04-21 CRAN (R 3.4.0)
##  methods     * 3.4.0   2017-06-05 local         
##  Rcpp          0.12.11 2017-05-22 CRAN (R 3.4.0)
##  rmarkdown     1.5     2017-04-26 CRAN (R 3.4.0)
##  rprojroot     1.2     2017-01-16 CRAN (R 3.4.0)
##  stats       * 3.4.0   2017-06-05 local         
##  stringi       1.1.5   2017-04-07 CRAN (R 3.4.0)
##  stringr       1.2.0   2017-02-18 CRAN (R 3.4.0)
##  tools         3.4.0   2017-06-05 local         
##  utils       * 3.4.0   2017-06-05 local         
##  withr         1.0.2   2016-06-20 CRAN (R 3.4.0)
##  yaml          2.1.14  2016-11-12 CRAN (R 3.4.0)
```

### Packages without Error

And here's my `session_info()` on a Linux machine with older package versions (although latest version via CRAN on Linux) that works fine and does not produce the same problem:

![image](https://user-images.githubusercontent.com/2837257/33952134-c617d6c4-dfe5-11e7-861d-d1beff95696e.png)

```
## Session info -------------------------------------------------------------
##  setting  value                       
##  version  R version 3.4.0 (2017-04-21)
##  system   x86_64, linux-gnu           
##  ui       X11                         
##  language (EN)                        
##  collate  en_US.UTF-8                 
##  tz       Etc/UTC                     
##  date     2017-12-13
## Packages -----------------------------------------------------------------
##  package     * version date       source        
##  backports     1.1.0   2017-05-22 CRAN (R 3.4.0)
##  base        * 3.4.0   2017-06-05 local         
##  compiler      3.4.0   2017-06-05 local         
##  datasets    * 3.4.0   2017-06-05 local         
##  devtools      1.13.2  2017-06-02 CRAN (R 3.4.0)
##  digest        0.6.12  2017-01-27 CRAN (R 3.4.0)
##  DT          * 0.2     2016-08-09 CRAN (R 3.4.0)
##  evaluate      0.10    2016-10-11 CRAN (R 3.4.0)
##  graphics    * 3.4.0   2017-06-05 local         
##  grDevices   * 3.4.0   2017-06-05 local         
##  htmltools     0.3.6   2017-04-28 CRAN (R 3.4.0)
##  htmlwidgets   0.8     2016-11-09 CRAN (R 3.4.0)
##  jsonlite      1.5     2017-06-01 CRAN (R 3.4.0)
##  knitr         1.16    2017-05-18 CRAN (R 3.4.0)
##  magrittr      1.5     2014-11-22 CRAN (R 3.4.0)
##  memoise       1.1.0   2017-04-21 CRAN (R 3.4.0)
##  methods     * 3.4.0   2017-06-05 local         
##  Rcpp          0.12.11 2017-05-22 CRAN (R 3.4.0)
##  rmarkdown     1.5     2017-04-26 CRAN (R 3.4.0)
##  rprojroot     1.2     2017-01-16 CRAN (R 3.4.0)
##  stats       * 3.4.0   2017-06-05 local         
##  stringi       1.1.5   2017-04-07 CRAN (R 3.4.0)
##  stringr       1.2.0   2017-02-18 CRAN (R 3.4.0)
##  tools         3.4.0   2017-06-05 local         
##  utils       * 3.4.0   2017-06-05 local         
##  withr         1.0.2   2016-06-20 CRAN (R 3.4.0)
##  yaml          2.1.14  2016-11-12 CRAN (R 3.4.0)
```

### Package differences

In reproducing this problem on a machine with older packages, the `DT` package is version 0.2 in both cases, so the problem must originate in another package, which I suspect is `rmarkdown` or `knitr`.

package       | old working | new broken
--------------|-------------|-----------
htmlwidgets   | 0.8         | 0.9
backports     | 1.1.0       | 1.1.1
yaml          | 2.1.14      | 2.1.15
Rcpp          | 0.12.11     | 0.12.14
stringi       | 1.1.5       | 1.1.6
**rmarkdown** | 1.5     | 1.8
**knitr**     | 1.16        | 1.17
evaluate      | 0.10        | 0.10.1

### Or is it pandoc / pdf generator?

In both cases `rmarkdown::pandoc_version()` produces the same version `1.17.2`.

#### Working on Linux

```bash
pdflatex --version
```

```
pdfTeX 3.14159265-2.6-1.40.15 (TeX Live 2015/dev/Debian)
kpathsea version 6.2.1dev
Copyright 2014 Peter Breitenlohner (eTeX)/Han The Thanh (pdfTeX).
There is NO warranty.  Redistribution of this software is
covered by the terms of both the pdfTeX copyright and
the Lesser GNU General Public License.
For more information about these matters, see the file
named COPYING and the pdfTeX source.
Primary author of pdfTeX: Peter Breitenlohner (eTeX)/Han The Thanh (pdfTeX).
Compiled with libpng 1.6.13; using libpng 1.6.13
Compiled with zlib 1.2.8; using zlib 1.2.8
Compiled with poppler version 0.26.5
```

#### Not working on Mac

```bash
pdflatex --version
```

This version is slightly newer (2016 vs 2015, *-1.40.17 vs *-1.40.15).

```
pdfTeX 3.14159265-2.6-1.40.17 (TeX Live 2016)
kpathsea version 6.2.2
Copyright 2016 Han The Thanh (pdfTeX) et al.
There is NO warranty.  Redistribution of this software is
covered by the terms of both the pdfTeX copyright and
the Lesser GNU General Public License.
For more information about these matters, see the file
named COPYING and the pdfTeX source.
Primary author of pdfTeX: Han The Thanh (pdfTeX) et al.
Compiled with libpng 1.6.21; using libpng 1.6.21
Compiled with zlib 1.2.8; using zlib 1.2.8
Compiled with xpdf version 3.04
```

Thanks in advance for any pointers :)

**Comment by yihui** [comment]
Thanks for the report! It is an exceptionally good bug report. I'll investigate it as soon as I have got a chance.

**Comment by yihui** [comment]
I tried for a while but could not reproduce your problem. I see several packages, including R itself, are outdated from your session info. To make it easier for me to debug, could you 
[first update everything?](https://yihui.name/en/2017/05/when-in-doubt-upgrade/) e.g. `update.packages(ask = FALSE, checkBuilt = TRUE)`.

BTW, it seems you don't use RStudio? I don't mean you should, but your Pandoc version seems to be low, too, and RStudio ships a more recent version.

**Comment by bbest** [comment]
Hi @yihui,

Thanks so much for looking into this. I am using RStudio (and absolutely love it!), went ahead and installed latest R 3.4.3 and RStudio 1.1.383, and saw the same problem. But looks like I pasted in the wrong session_info above for the problem on my Mac. Looking at the packages loaded again, I noticed:

- htmlwidgets   0.9   2017-11-17 Github (ramnathv/htmlwidgets@c042cca)

Uninstalling the Github snapshot of the htmlwidgets package (`remove.packages("htmlwidgets")`) and reinstalling from CRAN (`install.packages("htmlwidgets")`) fixed the problem 👍 

Sorry about the unnecessary hassle. Will ensure I'm properly updated with software next time before submitting issue 😉
